### ISSUE TITLE
strange part of code

### ISSUE THREAD

**Issue by kmdrGroch** [stryker-mutator/stryker#1133]
I have been looking through the code and I noticed some strange code:
```typescript
function isPrimitive(value: any) {
  return value !== 'object' && !Array.isArray(value);
}
```
shouldn't it be sth like:
```typescript
function isPrimitive(value: any) {
  return typeof value !== 'object' && !Array.isArray(value);
}
```
+ value can also be ``typeof function`` so all cases haven't been checked. (if you don't count NaN as a primitive then also NaN should be checked)

path: stryker\packages\stryker\test\helpers\producers.ts

so final code i suppose should be is like:
```typescript
function isPrimitive(value: any) {
  if(typeof value === 'number' && isNaN(value)) return false;
  if(typeof value !== 'object' && typeof value !== 'function') return true;
  return false;
}
```

**Comment by nicojs** [comment]
This is indeed a problem. Good proposal. Looking forward to a PR 👍 😎

**Comment by kmdrGroch** [comment]
@nicojs should i PR it? Or you will do it on hacktoberfest?

**Comment by nicojs** [comment]
Nope, this one isn't claimed yet. You're picking it up?

If you do + want a free T-shirt, make sure you're signing up for hacktoberfest: https://hacktoberfest.digitalocean.com/

**Comment by kmdrGroch** [comment]
I can pick it up :slightly_smiling_face:

**Comment by kmdrGroch** [comment]
@nicojs this can be closed.

**Comment by nicojs** [comment]
Yes it can. Thanks @kmdrGroch 

If you add `fixes #1133` in the description it will close this issue automatically when the PR is accepted.

**Comment by kmdrGroch** [comment]
thanks :D
